### PR TITLE
MAKE-1243: Implement gRPC service CreateTestResultsRecord

### PR DESCRIPTION
**Pull request by julianedwards**

### CODE REVIEW

File: rpc/internal/test_results.ext.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		TaskID:      t.TaskId,
+		Execution:   int(t.Execution),
+		RequestType: t.RequestType,
+		Mainline:    t.Mainline,
+	}
+}
+
+// Export exports TestResult to the corresponding TestResult type in the model
+// package.
+func (t TestResult) Export() (model.TestResult, error) {
```
Comment (tychoish) [inline rpc/internal/test_results.ext.go:26]: should this return a pointer?

File: rpc/internal/test_results_service.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+	record := model.CreateTestResults(info.Export(), conf.Bucket.TestResultsBucketType)
+	record.Setup(s.env)
+	return &TestResultsResponse{TestResultsRecordId: record.ID},
+		newRPCError(codes.Internal, errors.Wrap(record.SaveNew(ctx), "problem saving test results record"))
+}
+
+// AddTestResults adds test results to an existing test results record.
+func (s *testResultsService) AddTestResults(ctx context.Context, results *TestResults) (*TestResultsResponse, error) {
+	return nil, nil
```
Comment (tychoish) [inline rpc/internal/test_results_service.go:44]: can we return errors till tehy're implemented?

File: rpc/internal/test_results.ext.go (hunk 1 above)
Comment (julianedwards) [reply rpc/internal/test_results.ext.go:26]: No because the model functions don't take pointers

File: rpc/internal/test_results_service.go (hunk 2 above)
Comment (julianedwards) [reply rpc/internal/test_results_service.go:44]: sure

**Pull request merged**